code/controllers/subsystems/supply.dm

```diff
@@ -50,8 +75,399 @@ var/datum/controller/subsystem/cargo/SScargo
+		return 0
+	return charge_to_account(supply_account.account_number, "[commstation_name()] - Supply", "[charge_text]", "[commstation_name()] - Banking System", -charge_credits)
+//Gets the pending shipment costs for the items that are about to be shipped to the station
+/datum/controller/subsystem/cargo/proc/get_pending_shipment_cost(var/status="approved")
+	//Loop through all the orders marked as shipped and get the suppliers into a list of involved suppliers
+	var/list/suppliers = get_order_suppliers_by_status(status)
+	var/price = 0
+	for(var/supplier in suppliers)
+		var/datum/cargo_supplier/cs = SScargo.cargo_suppliers[supplier]
+		if(cs)
+			price += cs.shuttle_price
+	return price
+//Gets the pending shipment time for the items that are about to be shipped to the station
+/datum/controller/subsystem/cargo/proc/get_pending_shipment_time(var/status="approved")
+	//Loop through all the orders marked as shipped and get the suppliers into a list of involved suppliers
+	var/list/suppliers = get_order_suppliers_by_status(status)
+	var/time = 0
+	for(var/supplier in suppliers)
+		var/datum/cargo_supplier/cs = SScargo.cargo_suppliers[supplier]
+		if(cs)
+			time += cs.shuttle_time
+	return time
+
+/*
+	Shuttle Operations - Calling, Forcing, Canceling, Buying / Selling
+*/
+//Calls the shuttle. Returns a status message
+/datum/controller/subsystem/cargo/proc/shuttle_call(var/caller_name)
+	if(shuttle.at_station())
+		if (shuttle.forbidden_atoms_check())
+			. = "For safety reasons the automated supply shuttle cannot transport live organisms, classified nuclear weaponry or homing beacons."
+		else
+			movetime = 1200 //It always takes two minutes to get to centcom
+			shuttle.launch(src)
+			. = "Initiating launch sequence"
+			current_shipment.shuttle_recalled_by = caller_name
+	else
+		//Check if there is enough money in the cargo account for the current shipment
+		var/shipment_cost = get_pending_shipment_cost()
+		if( shipment_cost > get_cargo_money())
+			. = "The supply shuttle could not be called. Insufficient Funds. [shipment_cost] Credits required"
+		else
+			//Create a new shipment if one does not exist already
+			if(!current_shipment)
+				new_shipment()
+			
+			//Set the shuttle movement time
+			current_shipment.shuttle_time = get_pending_shipment_time()
+			current_shipment.shuttle_fee = shipment_cost
+
+			if(current_shipment.shuttle_time < 1200)
+				log_debug("Cargo: Shuttle Time less than 1200: [current_shipment.shuttle_time] - Setting to 1200")
+				current_shipment.shuttle_time = 1200
+
+			movetime = current_shipment.shuttle_time
+			//Launch it
+			shuttle.launch(src)
+			. = "The supply shuttle has been called and will arrive in approximately [round(SScargo.movetime/600,1)] minutes."
+			current_shipment.shuttle_called_by = caller_name
+	return .
+
+//Cancels the shuttle. Can return a status message
+/datum/controller/subsystem/cargo/proc/shuttle_cancel()
+	shuttle.cancel_launch(src)
+	return
```

Comment:
The return is there just for readability.
The proc can return a status message that is then displayed via the gui to the user.
It immediately clear that this proc does not return a status message